File: web/src/stores/session.js
import http from '@/js/axios'
import { defineStore } from 'pinia'

const useSessionStore = defineStore('session', {
	state: () => ({ 
		session: {}
	}),
	actions: {
		setSession(session) {
			this.session = session
		},

		async listSessions() {
			return await http.get('/api/sessions')
		},

		async createSession(session) {
			return await http.post('/api/sessions', session)
		},

		async getSession(id) {
			return await http.get(`/api/sessions/${id}`)
		},

		async deleteSession(id) {
			return await http.delete(`/api/sessions/${id}`)
		},

		async sendMessage(id, message) {
			return await http.post(`/api/sessions/${id}/messages`, message)
		},

		async getMessages(id) {
			return await http.get(`/api/sessions/${id}/messages`)
		}
	}
})

export default useSessionStore

File: api/session.go
package api

import (
	"net/http"

	"GIM/models"

	"github.com/gin-gonic/gin"
)

func listSessions(c *gin.Context) {
	id := c.GetString("userID")

	sessions, err := GE.Sessions.List(id)
	if err != nil {
		c.AbortWithStatusJSON(http.StatusInternalServerError, err)
		return
	}

	c.JSON(http.StatusOK, sessions)
}

func createSession(c *gin.Context) {
	var body models.Session
	if err := c.BindJSON(&body); err != nil {
		c.AbortWithStatusJSON(http.StatusBadRequest, err)
		return
	}

	body.OwnerID = c.GetString("userID")
	session, err := GE.Sessions.Create(&body)
	if err != nil {
		c.AbortWithStatusJSON(http.StatusInternalServerError, err)
		return
	}
	c.JSON(http.StatusOK, session)
}

func getSession(c *gin.Context) {
	var id string
	if err := SetPathVar(c, &id); err != nil {
		c.AbortWithStatusJSON(http.StatusBadRequest, err)
		return
	}

	session, err := GE.Sessions.Get(id)
	if err != nil {
		c.AbortWithStatusJSON(http.StatusInternalServerError, err)
		return
	}

	c.JSON(http.StatusOK, session)
}

func deleteSession(c *gin.Context) {
	var id string
	if err := SetPathVar(c, &id); err != nil {
		c.AbortWithStatusJSON(http.StatusBadRequest, err)
		return
	}

	if err := GE.Sessions.Delete(id); err != nil {
		c.AbortWithStatusJSON(http.StatusInternalServerError, err)
	}
}
